### PR TITLE
Ensure sonic yangs wheel is built before sonic-mgmt-common

### PR DESCRIPTION
**Pull request by sachinholla**
#### Why I did it

We plan to enhance UMF to import sonic yangs from the sonic-yang-models directory. Today UMF consumes yangs maintained in sonic-mgmt-commom repo only. UMF runs multiple code generators on these yangs during build time.  . sonic-yang-models also has build time templating. Hence for sonic-mgmt-common builds use the yangs from sonic-yang-models, the sonic_yang_models-1.0-py3-none-any.whl must be built fist.

Required for addressing the issue https://github.com/sonic-net/sonic-buildimage/issues/15183

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it

* Enhanced slave.mk to accept python wheels as dependency for a deb target. Dependent wheel names should be specified through the new {deb_name}_WHEEL_DEPENDS variable in the deb's make rules. The wheel will be built and installed in the slave docker before starting the deb build.

* Added sonic_yang_models-1.0-py3-none-any.whl as dependency for sonic-mgmt-common.deb. This is required for using the sonic yangs in UMF

#### How to verify it

Build only sonic-mgmt-common debian package as well as full image.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Comment by adyeung** [comment]
@qiluo-msft @xumia pls help review the change, this is needed to address the community issue #15183

**Review by kwangsuk — APPROVED** [review]

**Review by anand-kumar-subramanian — APPROVED** [review]

**Comment by adyeung** [comment]
@qiluo-msft @xumia got review approval from Kwan and Anand, will merge tomorrow if there is no further comments

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**
